### PR TITLE
Add and use 'testAccAvailableEc2InstanceTypeForAvailabilityZone' config generator

### PR DESCRIPTION
**Pull request by ewbankkit**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Closes https://github.com/terraform-providers/terraform-provider-aws/issues/13083.

Release note for [CHANGELOG](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md):

```release-note
NONE
```

Output from acceptance testing:

```console
$ make testacc TEST=./aws/ TESTARGS='-run=TestAccAWSEIP_Instance_Reassociate'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws/ -v -count 1 -parallel 20 -run=TestAccAWSEIP_Instance_Reassociate -timeout 120m
=== RUN   TestAccAWSEIP_Instance_Reassociate
=== PAUSE TestAccAWSEIP_Instance_Reassociate
=== CONT  TestAccAWSEIP_Instance_Reassociate
--- PASS: TestAccAWSEIP_Instance_Reassociate (188.11s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	188.142s
$ make testacc TEST=./aws/ TESTARGS='-run=TestAccAWSAMIFromInstance_'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws/ -v -count 1 -parallel 20 -run=TestAccAWSAMIFromInstance_ -timeout 120m
=== RUN   TestAccAWSAMIFromInstance_basic
=== PAUSE TestAccAWSAMIFromInstance_basic
=== RUN   TestAccAWSAMIFromInstance_tags
=== PAUSE TestAccAWSAMIFromInstance_tags
=== CONT  TestAccAWSAMIFromInstance_basic
=== CONT  TestAccAWSAMIFromInstance_tags
--- PASS: TestAccAWSAMIFromInstance_basic (407.63s)
--- PASS: TestAccAWSAMIFromInstance_tags (422.76s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	422.794s
$ make testacc TEST=./aws/ TESTARGS='-run=TestAccAWSCloudWatchMetricAlarm_AlarmActions_EC2Automate'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws/ -v -count 1 -parallel 20 -run=TestAccAWSCloudWatchMetricAlarm_AlarmActions_EC2Automate -timeout 120m
=== RUN   TestAccAWSCloudWatchMetricAlarm_AlarmActions_EC2Automate
=== PAUSE TestAccAWSCloudWatchMetricAlarm_AlarmActions_EC2Automate
=== CONT  TestAccAWSCloudWatchMetricAlarm_AlarmActions_EC2Automate
--- PASS: TestAccAWSCloudWatchMetricAlarm_AlarmActions_EC2Automate (240.68s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	240.738s
$ make testacc TEST=./aws/ TESTARGS='-run=TestAccAWSStorageGatewayGateway_SmbActiveDirectorySettings'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws/ -v -count 1 -parallel 20 -run=TestAccAWSStorageGatewayGateway_SmbActiveDirectorySettings -timeout 120m
=== RUN   TestAccAWSStorageGatewayGateway_SmbActiveDirectorySettings
=== PAUSE TestAccAWSStorageGatewayGateway_SmbActiveDirectorySettings
=== CONT  TestAccAWSStorageGatewayGateway_SmbActiveDirectorySettings
--- PASS: TestAccAWSStorageGatewayGateway_SmbActiveDirectorySettings (758.96s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	759.004s
$ make testacc TEST=./aws/ TESTARGS='-run=TestAccAWSStorageGatewayGateway_GatewayType_FileS3'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws/ -v -count 1 -parallel 20 -run=TestAccAWSStorageGatewayGateway_GatewayType_FileS3 -timeout 120m
=== RUN   TestAccAWSStorageGatewayGateway_GatewayType_FileS3
=== PAUSE TestAccAWSStorageGatewayGateway_GatewayType_FileS3
=== CONT  TestAccAWSStorageGatewayGateway_GatewayType_FileS3
--- PASS: TestAccAWSStorageGatewayGateway_GatewayType_FileS3 (293.30s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	293.334s
$ make testacc TEST=./aws/ TESTARGS='-run=TestAccAWSStorageGatewayGateway_GatewayType_Cached'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws/ -v -count 1 -parallel 20 -run=TestAccAWSStorageGatewayGateway_GatewayType_Cached -timeout 120m
=== RUN   TestAccAWSStorageGatewayGateway_GatewayType_Cached
=== PAUSE TestAccAWSStorageGatewayGateway_GatewayType_Cached
=== CONT  TestAccAWSStorageGatewayGateway_GatewayType_Cached
--- PASS: TestAccAWSStorageGatewayGateway_GatewayType_Cached (267.94s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	267.986s
$ make testacc TEST=./aws/ TESTARGS='-run=TestAccAWSSpotFleetRequest_'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws/ -v -count 1 -parallel 2 -run=TestAccAWSSpotFleetRequest_ -timeout 120m
=== RUN   TestAccAWSSpotFleetRequest_basic
=== PAUSE TestAccAWSSpotFleetRequest_basic
=== RUN   TestAccAWSSpotFleetRequest_tags
=== PAUSE TestAccAWSSpotFleetRequest_tags
=== RUN   TestAccAWSSpotFleetRequest_associatePublicIpAddress
=== PAUSE TestAccAWSSpotFleetRequest_associatePublicIpAddress
=== RUN   TestAccAWSSpotFleetRequest_launchTemplate
=== PAUSE TestAccAWSSpotFleetRequest_launchTemplate
=== RUN   TestAccAWSSpotFleetRequest_launchTemplate_multiple
=== PAUSE TestAccAWSSpotFleetRequest_launchTemplate_multiple
=== RUN   TestAccAWSSpotFleetRequest_launchTemplateWithOverrides
=== PAUSE TestAccAWSSpotFleetRequest_launchTemplateWithOverrides
=== RUN   TestAccAWSSpotFleetRequest_launchTemplateToLaunchSpec
--- PASS: TestAccAWSSpotFleetRequest_launchTemplateToLaunchSpec (491.67s)
=== RUN   TestAccAWSSpotFleetRequest_launchSpecToLaunchTemplate
--- PASS: TestAccAWSSpotFleetRequest_launchSpecToLaunchTemplate (459.29s)
=== RUN   TestAccAWSSpotFleetRequest_instanceInterruptionBehavior
=== PAUSE TestAccAWSSpotFleetRequest_instanceInterruptionBehavior
=== RUN   TestAccAWSSpotFleetRequest_fleetType
=== PAUSE TestAccAWSSpotFleetRequest_fleetType
=== RUN   TestAccAWSSpotFleetRequest_iamInstanceProfileArn
=== PAUSE TestAccAWSSpotFleetRequest_iamInstanceProfileArn
=== RUN   TestAccAWSSpotFleetRequest_changePriceForcesNewRequest
=== PAUSE TestAccAWSSpotFleetRequest_changePriceForcesNewRequest
=== RUN   TestAccAWSSpotFleetRequest_updateTargetCapacity
=== PAUSE TestAccAWSSpotFleetRequest_updateTargetCapacity
=== RUN   TestAccAWSSpotFleetRequest_updateExcessCapacityTerminationPolicy
=== PAUSE TestAccAWSSpotFleetRequest_updateExcessCapacityTerminationPolicy
=== RUN   TestAccAWSSpotFleetRequest_lowestPriceAzOrSubnetInRegion
=== PAUSE TestAccAWSSpotFleetRequest_lowestPriceAzOrSubnetInRegion
=== RUN   TestAccAWSSpotFleetRequest_lowestPriceAzInGivenList
=== PAUSE TestAccAWSSpotFleetRequest_lowestPriceAzInGivenList
=== RUN   TestAccAWSSpotFleetRequest_lowestPriceSubnetInGivenList
=== PAUSE TestAccAWSSpotFleetRequest_lowestPriceSubnetInGivenList
=== RUN   TestAccAWSSpotFleetRequest_multipleInstanceTypesInSameAz
=== PAUSE TestAccAWSSpotFleetRequest_multipleInstanceTypesInSameAz
=== RUN   TestAccAWSSpotFleetRequest_multipleInstanceTypesInSameSubnet
=== PAUSE TestAccAWSSpotFleetRequest_multipleInstanceTypesInSameSubnet
=== RUN   TestAccAWSSpotFleetRequest_overriddingSpotPrice
=== PAUSE TestAccAWSSpotFleetRequest_overriddingSpotPrice
=== RUN   TestAccAWSSpotFleetRequest_withoutSpotPrice
=== PAUSE TestAccAWSSpotFleetRequest_withoutSpotPrice
=== RUN   TestAccAWSSpotFleetRequest_diversifiedAllocation
=== PAUSE TestAccAWSSpotFleetRequest_diversifiedAllocation
=== RUN   TestAccAWSSpotFleetRequest_multipleInstancePools
=== PAUSE TestAccAWSSpotFleetRequest_multipleInstancePools
=== RUN   TestAccAWSSpotFleetRequest_withWeightedCapacity
=== PAUSE TestAccAWSSpotFleetRequest_withWeightedCapacity
=== RUN   TestAccAWSSpotFleetRequest_withEBSDisk
=== PAUSE TestAccAWSSpotFleetRequest_withEBSDisk
=== RUN   TestAccAWSSpotFleetRequest_LaunchSpecification_EbsBlockDevice_KmsKeyId
=== PAUSE TestAccAWSSpotFleetRequest_LaunchSpecification_EbsBlockDevice_KmsKeyId
=== RUN   TestAccAWSSpotFleetRequest_LaunchSpecification_RootBlockDevice_KmsKeyId
=== PAUSE TestAccAWSSpotFleetRequest_LaunchSpecification_RootBlockDevice_KmsKeyId
=== RUN   TestAccAWSSpotFleetRequest_withTags
=== PAUSE TestAccAWSSpotFleetRequest_withTags
=== RUN   TestAccAWSSpotFleetRequest_placementTenancyAndGroup
=== PAUSE TestAccAWSSpotFleetRequest_placementTenancyAndGroup
=== RUN   TestAccAWSSpotFleetRequest_WithELBs
=== PAUSE TestAccAWSSpotFleetRequest_WithELBs
=== RUN   TestAccAWSSpotFleetRequest_WithTargetGroups
=== PAUSE TestAccAWSSpotFleetRequest_WithTargetGroups
=== RUN   TestAccAWSSpotFleetRequest_WithInstanceStoreAmi
--- SKIP: TestAccAWSSpotFleetRequest_WithInstanceStoreAmi (0.00s)
    resource_aws_spot_fleet_request_test.go:881: Test fails due to test harness constraints
=== RUN   TestAccAWSSpotFleetRequest_disappears
=== PAUSE TestAccAWSSpotFleetRequest_disappears
=== CONT  TestAccAWSSpotFleetRequest_basic
=== CONT  TestAccAWSSpotFleetRequest_disappears
--- PASS: TestAccAWSSpotFleetRequest_basic (275.55s)
=== CONT  TestAccAWSSpotFleetRequest_WithTargetGroups
--- PASS: TestAccAWSSpotFleetRequest_disappears (283.04s)
=== CONT  TestAccAWSSpotFleetRequest_WithELBs
--- PASS: TestAccAWSSpotFleetRequest_WithELBs (309.71s)
=== CONT  TestAccAWSSpotFleetRequest_placementTenancyAndGroup
--- PASS: TestAccAWSSpotFleetRequest_placementTenancyAndGroup (71.81s)
=== CONT  TestAccAWSSpotFleetRequest_withTags
--- PASS: TestAccAWSSpotFleetRequest_WithTargetGroups (464.89s)
=== CONT  TestAccAWSSpotFleetRequest_LaunchSpecification_RootBlockDevice_KmsKeyId
--- PASS: TestAccAWSSpotFleetRequest_LaunchSpecification_RootBlockDevice_KmsKeyId (148.77s)
=== CONT  TestAccAWSSpotFleetRequest_LaunchSpecification_EbsBlockDevice_KmsKeyId
--- PASS: TestAccAWSSpotFleetRequest_withTags (348.56s)
=== CONT  TestAccAWSSpotFleetRequest_withEBSDisk
--- PASS: TestAccAWSSpotFleetRequest_LaunchSpecification_EbsBlockDevice_KmsKeyId (149.58s)
=== CONT  TestAccAWSSpotFleetRequest_withWeightedCapacity
--- PASS: TestAccAWSSpotFleetRequest_withEBSDisk (276.36s)
=== CONT  TestAccAWSSpotFleetRequest_multipleInstancePools
--- PASS: TestAccAWSSpotFleetRequest_withWeightedCapacity (305.96s)
=== CONT  TestAccAWSSpotFleetRequest_diversifiedAllocation
--- PASS: TestAccAWSSpotFleetRequest_multipleInstancePools (322.14s)
=== CONT  TestAccAWSSpotFleetRequest_withoutSpotPrice
--- PASS: TestAccAWSSpotFleetRequest_diversifiedAllocation (278.79s)
=== CONT  TestAccAWSSpotFleetRequest_overriddingSpotPrice
--- PASS: TestAccAWSSpotFleetRequest_withoutSpotPrice (286.55s)
=== CONT  TestAccAWSSpotFleetRequest_multipleInstanceTypesInSameSubnet
--- PASS: TestAccAWSSpotFleetRequest_overriddingSpotPrice (277.84s)
=== CONT  TestAccAWSSpotFleetRequest_multipleInstanceTypesInSameAz
--- PASS: TestAccAWSSpotFleetRequest_multipleInstanceTypesInSameSubnet (274.61s)
=== CONT  TestAccAWSSpotFleetRequest_lowestPriceSubnetInGivenList
--- PASS: TestAccAWSSpotFleetRequest_multipleInstanceTypesInSameAz (277.08s)
=== CONT  TestAccAWSSpotFleetRequest_lowestPriceAzInGivenList
--- PASS: TestAccAWSSpotFleetRequest_lowestPriceSubnetInGivenList (278.49s)
=== CONT  TestAccAWSSpotFleetRequest_lowestPriceAzOrSubnetInRegion
--- PASS: TestAccAWSSpotFleetRequest_lowestPriceAzInGivenList (330.22s)
=== CONT  TestAccAWSSpotFleetRequest_updateExcessCapacityTerminationPolicy
--- PASS: TestAccAWSSpotFleetRequest_lowestPriceAzOrSubnetInRegion (246.91s)
=== CONT  TestAccAWSSpotFleetRequest_updateTargetCapacity
--- PASS: TestAccAWSSpotFleetRequest_updateExcessCapacityTerminationPolicy (655.89s)
=== CONT  TestAccAWSSpotFleetRequest_changePriceForcesNewRequest
--- PASS: TestAccAWSSpotFleetRequest_updateTargetCapacity (806.86s)
=== CONT  TestAccAWSSpotFleetRequest_iamInstanceProfileArn
--- PASS: TestAccAWSSpotFleetRequest_changePriceForcesNewRequest (594.02s)
=== CONT  TestAccAWSSpotFleetRequest_fleetType
--- PASS: TestAccAWSSpotFleetRequest_iamInstanceProfileArn (274.36s)
=== CONT  TestAccAWSSpotFleetRequest_instanceInterruptionBehavior
--- PASS: TestAccAWSSpotFleetRequest_fleetType (267.13s)
=== CONT  TestAccAWSSpotFleetRequest_launchTemplateWithOverrides
--- PASS: TestAccAWSSpotFleetRequest_instanceInterruptionBehavior (247.06s)
=== CONT  TestAccAWSSpotFleetRequest_launchTemplate_multiple
--- PASS: TestAccAWSSpotFleetRequest_launchTemplateWithOverrides (276.99s)
=== CONT  TestAccAWSSpotFleetRequest_launchTemplate
--- PASS: TestAccAWSSpotFleetRequest_launchTemplate_multiple (277.20s)
=== CONT  TestAccAWSSpotFleetRequest_associatePublicIpAddress
--- PASS: TestAccAWSSpotFleetRequest_associatePublicIpAddress (265.37s)
=== CONT  TestAccAWSSpotFleetRequest_tags
--- PASS: TestAccAWSSpotFleetRequest_launchTemplate (269.71s)
--- PASS: TestAccAWSSpotFleetRequest_tags (331.15s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	5851.188s
```

### CODE REVIEW

**Comment by ewbankkit** [comment]
Removing _WIP_.
Ready for review.

**Comment by ewbankkit** [comment]
Rebased to remove merge conflict and adapted `aws_volume_attachment` resource acceptance tests.

```console
$ make testacc TEST=./aws/ TESTARGS='-run=TestAccAWSVolumeAttachment_'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws/ -v -count 1 -parallel 20 -run=TestAccAWSVolumeAttachment_ -timeout 120m
=== RUN   TestAccAWSVolumeAttachment_basic
=== PAUSE TestAccAWSVolumeAttachment_basic
=== RUN   TestAccAWSVolumeAttachment_skipDestroy
=== PAUSE TestAccAWSVolumeAttachment_skipDestroy
=== RUN   TestAccAWSVolumeAttachment_attachStopped
=== PAUSE TestAccAWSVolumeAttachment_attachStopped
=== RUN   TestAccAWSVolumeAttachment_update
=== PAUSE TestAccAWSVolumeAttachment_update
=== RUN   TestAccAWSVolumeAttachment_disappears
=== PAUSE TestAccAWSVolumeAttachment_disappears
=== CONT  TestAccAWSVolumeAttachment_basic
=== CONT  TestAccAWSVolumeAttachment_update
=== CONT  TestAccAWSVolumeAttachment_attachStopped
=== CONT  TestAccAWSVolumeAttachment_disappears
=== CONT  TestAccAWSVolumeAttachment_skipDestroy
--- PASS: TestAccAWSVolumeAttachment_disappears (106.67s)
--- PASS: TestAccAWSVolumeAttachment_skipDestroy (135.96s)
--- PASS: TestAccAWSVolumeAttachment_basic (136.05s)
--- PASS: TestAccAWSVolumeAttachment_attachStopped (160.20s)
--- PASS: TestAccAWSVolumeAttachment_update (162.13s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	162.213s
```

**Comment by ewbankkit** [comment]
Rebased to remove merge conflicts.
Re-ran acceptance tests:

```console
$ make testacc TEST=./aws/ TESTARGS='-run=TestAccAWSEIP_Instance_Reassociate'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws -v -count 1 -parallel 20 -run=TestAccAWSEIP_Instance_Reassociate -timeout 120m
=== RUN   TestAccAWSEIP_Instance_Reassociate
=== PAUSE TestAccAWSEIP_Instance_Reassociate
=== CONT  TestAccAWSEIP_Instance_Reassociate
--- PASS: TestAccAWSEIP_Instance_Reassociate (114.28s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	114.340s
$ make testacc TEST=./aws/ TESTARGS='-run=TestAccAWSCloudWatchMetricAlarm_AlarmActions_EC2Automate'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws -v -count 1 -parallel 20 -run=TestAccAWSCloudWatchMetricAlarm_AlarmActions_EC2Automate -timeout 120m
=== RUN   TestAccAWSCloudWatchMetricAlarm_AlarmActions_EC2Automate
=== PAUSE TestAccAWSCloudWatchMetricAlarm_AlarmActions_EC2Automate
=== CONT  TestAccAWSCloudWatchMetricAlarm_AlarmActions_EC2Automate
--- PASS: TestAccAWSCloudWatchMetricAlarm_AlarmActions_EC2Automate (159.57s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	159.622s
```

File: aws/resource_aws_spot_fleet_request_test.go [hunk 1]
```diff
@@ -1504,8 +1466,8 @@ resource "aws_launch_template" "test" {
 }
 
 resource "aws_spot_fleet_request" "test" {
-  iam_fleet_role                      = "${aws_iam_role.test-role.arn}"
-  spot_price                          = "0.05"
+  iam_fleet_role                      = "${aws_iam_role.test.arn}"
+  spot_price                          = "0.005"
```
Comment (YakDriver) [inline aws/resource_aws_spot_fleet_request_test.go:1470]: Please leave the prices as is until we move to a pricing data source.

File: aws/resource_aws_instance_test.go [hunk 2]
```diff
@@ -5131,6 +5132,23 @@ resource "aws_instance" "test" {
 `, rName))
 }
 
+const testAccAwsEc2InstanceConfigDynamicEBSBlockDevices = `
+resource "aws_instance" "test" {
+  ami           = "ami-55a7ea65"
```
Comment (YakDriver) [inline aws/resource_aws_instance_test.go:5137]: Avoid using hardcoded AMI IDs. Use `testAccLatestAmazonLinuxHvmEbsAmiConfig()` and `data.aws_ami.amzn-ami-minimal-hvm-ebs.id` instread.

File: aws/resource_aws_spot_fleet_request_test.go (hunk 1 above)
Comment (ewbankkit) [reply aws/resource_aws_spot_fleet_request_test.go:1470]: Will do; Got overeager during the rebase.

File: aws/resource_aws_instance_test.go (hunk 2 above)
Comment (ewbankkit) [reply aws/resource_aws_instance_test.go:5137]: Will do; That's fixed over in https://github.com/terraform-providers/terraform-provider-aws/pull/14159 and I'll do the same here.

Comment (ewbankkit) [reply aws/resource_aws_instance_test.go]: Done.
Relevant acceptance test:

```console
$ make testacc TEST=./aws/ TESTARGS='-run=TestAccAWSInstance_EbsRootDevice_MultipleDynamicEBSBlockDevices'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws -v -count 1 -parallel 2 -run=TestAccAWSInstance_EbsRootDevice_MultipleDynamicEBSBlockDevices -timeout 120m
=== RUN   TestAccAWSInstance_EbsRootDevice_MultipleDynamicEBSBlockDevices
=== PAUSE TestAccAWSInstance_EbsRootDevice_MultipleDynamicEBSBlockDevices
=== CONT  TestAccAWSInstance_EbsRootDevice_MultipleDynamicEBSBlockDevices
--- PASS: TestAccAWSInstance_EbsRootDevice_MultipleDynamicEBSBlockDevices (211.77s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	211.825s
```

File: aws/resource_aws_spot_fleet_request_test.go (hunk 1 above)
Comment (ewbankkit) [reply aws/resource_aws_spot_fleet_request_test.go]: Done.

**Comment by ewbankkit** [comment]
Re-ran spot fleet request acceptance tests:

```console
$ make testacc TEST=./aws/ TESTARGS='-run=TestAccAWSSpotFleetRequest_'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws -v -count 1 -parallel 2 -run=TestAccAWSSpotFleetRequest_ -timeout 120m
=== RUN   TestAccAWSSpotFleetRequest_basic
=== PAUSE TestAccAWSSpotFleetRequest_basic
=== RUN   TestAccAWSSpotFleetRequest_tags
=== PAUSE TestAccAWSSpotFleetRequest_tags
=== RUN   TestAccAWSSpotFleetRequest_associatePublicIpAddress
=== PAUSE TestAccAWSSpotFleetRequest_associatePublicIpAddress
=== RUN   TestAccAWSSpotFleetRequest_launchTemplate
=== PAUSE TestAccAWSSpotFleetRequest_launchTemplate
=== RUN   TestAccAWSSpotFleetRequest_launchTemplate_multiple
=== PAUSE TestAccAWSSpotFleetRequest_launchTemplate_multiple
=== RUN   TestAccAWSSpotFleetRequest_launchTemplateWithOverrides
=== PAUSE TestAccAWSSpotFleetRequest_launchTemplateWithOverrides
=== RUN   TestAccAWSSpotFleetRequest_launchTemplateToLaunchSpec
=== PAUSE TestAccAWSSpotFleetRequest_launchTemplateToLaunchSpec
=== RUN   TestAccAWSSpotFleetRequest_launchSpecToLaunchTemplate
=== PAUSE TestAccAWSSpotFleetRequest_launchSpecToLaunchTemplate
=== RUN   TestAccAWSSpotFleetRequest_instanceInterruptionBehavior
=== PAUSE TestAccAWSSpotFleetRequest_instanceInterruptionBehavior
=== RUN   TestAccAWSSpotFleetRequest_fleetType
=== PAUSE TestAccAWSSpotFleetRequest_fleetType
=== RUN   TestAccAWSSpotFleetRequest_iamInstanceProfileArn
=== PAUSE TestAccAWSSpotFleetRequest_iamInstanceProfileArn
=== RUN   TestAccAWSSpotFleetRequest_changePriceForcesNewRequest
=== PAUSE TestAccAWSSpotFleetRequest_changePriceForcesNewRequest
=== RUN   TestAccAWSSpotFleetRequest_updateTargetCapacity
=== PAUSE TestAccAWSSpotFleetRequest_updateTargetCapacity
=== RUN   TestAccAWSSpotFleetRequest_updateExcessCapacityTerminationPolicy
=== PAUSE TestAccAWSSpotFleetRequest_updateExcessCapacityTerminationPolicy
=== RUN   TestAccAWSSpotFleetRequest_lowestPriceAzOrSubnetInRegion
=== PAUSE TestAccAWSSpotFleetRequest_lowestPriceAzOrSubnetInRegion
=== RUN   TestAccAWSSpotFleetRequest_lowestPriceAzInGivenList
=== PAUSE TestAccAWSSpotFleetRequest_lowestPriceAzInGivenList
=== RUN   TestAccAWSSpotFleetRequest_lowestPriceSubnetInGivenList
=== PAUSE TestAccAWSSpotFleetRequest_lowestPriceSubnetInGivenList
=== RUN   TestAccAWSSpotFleetRequest_multipleInstanceTypesInSameAz
=== PAUSE TestAccAWSSpotFleetRequest_multipleInstanceTypesInSameAz
=== RUN   TestAccAWSSpotFleetRequest_multipleInstanceTypesInSameSubnet
=== PAUSE TestAccAWSSpotFleetRequest_multipleInstanceTypesInSameSubnet
=== RUN   TestAccAWSSpotFleetRequest_overriddingSpotPrice
=== PAUSE TestAccAWSSpotFleetRequest_overriddingSpotPrice
=== RUN   TestAccAWSSpotFleetRequest_withoutSpotPrice
=== PAUSE TestAccAWSSpotFleetRequest_withoutSpotPrice
=== RUN   TestAccAWSSpotFleetRequest_diversifiedAllocation
=== PAUSE TestAccAWSSpotFleetRequest_diversifiedAllocation
=== RUN   TestAccAWSSpotFleetRequest_multipleInstancePools
=== PAUSE TestAccAWSSpotFleetRequest_multipleInstancePools
=== RUN   TestAccAWSSpotFleetRequest_withWeightedCapacity
=== PAUSE TestAccAWSSpotFleetRequest_withWeightedCapacity
=== RUN   TestAccAWSSpotFleetRequest_withEBSDisk
=== PAUSE TestAccAWSSpotFleetRequest_withEBSDisk
=== RUN   TestAccAWSSpotFleetRequest_LaunchSpecification_EbsBlockDevice_KmsKeyId
=== PAUSE TestAccAWSSpotFleetRequest_LaunchSpecification_EbsBlockDevice_KmsKeyId
=== RUN   TestAccAWSSpotFleetRequest_LaunchSpecification_RootBlockDevice_KmsKeyId
=== PAUSE TestAccAWSSpotFleetRequest_LaunchSpecification_RootBlockDevice_KmsKeyId
=== RUN   TestAccAWSSpotFleetRequest_withTags
=== PAUSE TestAccAWSSpotFleetRequest_withTags
=== RUN   TestAccAWSSpotFleetRequest_placementTenancyAndGroup
=== PAUSE TestAccAWSSpotFleetRequest_placementTenancyAndGroup
=== RUN   TestAccAWSSpotFleetRequest_WithELBs
=== PAUSE TestAccAWSSpotFleetRequest_WithELBs
=== RUN   TestAccAWSSpotFleetRequest_WithTargetGroups
=== PAUSE TestAccAWSSpotFleetRequest_WithTargetGroups
=== RUN   TestAccAWSSpotFleetRequest_WithInstanceStoreAmi
--- SKIP: TestAccAWSSpotFleetRequest_WithInstanceStoreAmi (0.00s)
    resource_aws_spot_fleet_request_test.go:1076: Test fails due to test harness constraints
=== RUN   TestAccAWSSpotFleetRequest_disappears
=== PAUSE TestAccAWSSpotFleetRequest_disappears
=== CONT  TestAccAWSSpotFleetRequest_basic
=== CONT  TestAccAWSSpotFleetRequest_multipleInstanceTypesInSameAz
--- PASS: TestAccAWSSpotFleetRequest_multipleInstanceTypesInSameAz (273.52s)
=== CONT  TestAccAWSSpotFleetRequest_disappears
--- PASS: TestAccAWSSpotFleetRequest_basic (274.21s)
=== CONT  TestAccAWSSpotFleetRequest_WithTargetGroups
--- PASS: TestAccAWSSpotFleetRequest_disappears (356.73s)
=== CONT  TestAccAWSSpotFleetRequest_WithELBs
--- PASS: TestAccAWSSpotFleetRequest_WithTargetGroups (389.64s)
=== CONT  TestAccAWSSpotFleetRequest_placementTenancyAndGroup
--- PASS: TestAccAWSSpotFleetRequest_placementTenancyAndGroup (61.61s)
=== CONT  TestAccAWSSpotFleetRequest_withTags
--- PASS: TestAccAWSSpotFleetRequest_WithELBs (329.12s)
=== CONT  TestAccAWSSpotFleetRequest_LaunchSpecification_RootBlockDevice_KmsKeyId
--- PASS: TestAccAWSSpotFleetRequest_withTags (250.34s)
=== CONT  TestAccAWSSpotFleetRequest_LaunchSpecification_EbsBlockDevice_KmsKeyId
--- PASS: TestAccAWSSpotFleetRequest_LaunchSpecification_RootBlockDevice_KmsKeyId (152.33s)
=== CONT  TestAccAWSSpotFleetRequest_withEBSDisk
--- PASS: TestAccAWSSpotFleetRequest_LaunchSpecification_EbsBlockDevice_KmsKeyId (141.08s)
=== CONT  TestAccAWSSpotFleetRequest_withWeightedCapacity
--- PASS: TestAccAWSSpotFleetRequest_withEBSDisk (273.58s)
=== CONT  TestAccAWSSpotFleetRequest_multipleInstancePools
--- PASS: TestAccAWSSpotFleetRequest_withWeightedCapacity (286.36s)
=== CONT  TestAccAWSSpotFleetRequest_diversifiedAllocation
--- PASS: TestAccAWSSpotFleetRequest_diversifiedAllocation (281.00s)
=== CONT  TestAccAWSSpotFleetRequest_withoutSpotPrice
--- PASS: TestAccAWSSpotFleetRequest_multipleInstancePools (298.99s)
=== CONT  TestAccAWSSpotFleetRequest_overriddingSpotPrice
--- PASS: TestAccAWSSpotFleetRequest_overriddingSpotPrice (285.70s)
=== CONT  TestAccAWSSpotFleetRequest_multipleInstanceTypesInSameSubnet
--- PASS: TestAccAWSSpotFleetRequest_withoutSpotPrice (285.80s)
=== CONT  TestAccAWSSpotFleetRequest_fleetType
--- PASS: TestAccAWSSpotFleetRequest_multipleInstanceTypesInSameSubnet (201.77s)
=== CONT  TestAccAWSSpotFleetRequest_lowestPriceSubnetInGivenList
--- PASS: TestAccAWSSpotFleetRequest_fleetType (323.52s)
=== CONT  TestAccAWSSpotFleetRequest_lowestPriceAzInGivenList
--- PASS: TestAccAWSSpotFleetRequest_lowestPriceSubnetInGivenList (286.32s)
=== CONT  TestAccAWSSpotFleetRequest_lowestPriceAzOrSubnetInRegion
--- PASS: TestAccAWSSpotFleetRequest_lowestPriceAzInGivenList (244.69s)
=== CONT  TestAccAWSSpotFleetRequest_updateExcessCapacityTerminationPolicy
--- PASS: TestAccAWSSpotFleetRequest_lowestPriceAzOrSubnetInRegion (252.36s)
=== CONT  TestAccAWSSpotFleetRequest_updateTargetCapacity
--- PASS: TestAccAWSSpotFleetRequest_updateExcessCapacityTerminationPolicy (511.70s)
=== CONT  TestAccAWSSpotFleetRequest_changePriceForcesNewRequest
--- PASS: TestAccAWSSpotFleetRequest_updateTargetCapacity (760.41s)
=== CONT  TestAccAWSSpotFleetRequest_iamInstanceProfileArn
--- PASS: TestAccAWSSpotFleetRequest_changePriceForcesNewRequest (564.99s)
=== CONT  TestAccAWSSpotFleetRequest_launchTemplateWithOverrides
--- PASS: TestAccAWSSpotFleetRequest_iamInstanceProfileArn (275.08s)
=== CONT  TestAccAWSSpotFleetRequest_instanceInterruptionBehavior
--- PASS: TestAccAWSSpotFleetRequest_launchTemplateWithOverrides (202.16s)
=== CONT  TestAccAWSSpotFleetRequest_launchSpecToLaunchTemplate
--- PASS: TestAccAWSSpotFleetRequest_instanceInterruptionBehavior (284.50s)
=== CONT  TestAccAWSSpotFleetRequest_launchTemplateToLaunchSpec
--- PASS: TestAccAWSSpotFleetRequest_launchSpecToLaunchTemplate (564.43s)
=== CONT  TestAccAWSSpotFleetRequest_launchTemplate
--- PASS: TestAccAWSSpotFleetRequest_launchTemplateToLaunchSpec (568.03s)
=== CONT  TestAccAWSSpotFleetRequest_launchTemplate_multiple
--- PASS: TestAccAWSSpotFleetRequest_launchTemplate (274.88s)
=== CONT  TestAccAWSSpotFleetRequest_associatePublicIpAddress
--- PASS: TestAccAWSSpotFleetRequest_launchTemplate_multiple (270.57s)
=== CONT  TestAccAWSSpotFleetRequest_tags
--- PASS: TestAccAWSSpotFleetRequest_associatePublicIpAddress (262.48s)
--- PASS: TestAccAWSSpotFleetRequest_tags (266.34s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	5135.401s
```

**Comment by YakDriver** [comment]
This looks good, @ewbankkit !

**Pull request merged**